### PR TITLE
feat(events): EventBus policy

### PR DESCRIPTION
**Pull request by otaviomacedo**
Allow users to set the resource policy for an `EventBus`.

Closes https://github.com/aws/aws-cdk/issues/23021.

----

### All Submissions:

* [ ] Have you followed the guidelines in our [Contributing guide?](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md)

### Adding new Construct Runtime Dependencies:

* [ ] This PR adds new construct runtime dependencies following the process described [here](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md/#adding-construct-runtime-dependencies)

### New Features

* [ ] Have you added the new feature to an [integration test](https://github.com/aws/aws-cdk/blob/main/INTEGRATION_TESTS.md)?
	* [ ] Did you use `yarn integ` to deploy the infrastructure and generate the snapshot (i.e. `yarn integ` without `--dry-run`)?

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk/aws-events/lib/event-bus.ts [hunk 1]
```diff
@@ -332,13 +338,39 @@ export class EventBus extends EventBusBase {
...
+    if (statement.sid == null) {
+      throw new Error('Event Bus policy statements must have a sid');
+    }
+
+    if (this.policy) {
+      // The policy can contain only one statement
+      return { statementAdded: false };
+    }
+
+    if (this.autoCreatePolicy) {
```
Comment (Naumel) [inline packages/@aws-cdk/aws-events/lib/event-bus.ts:352]: [nit] In the context of this class alone, it reads strangely to check for an `autoCreatePolicy` in the `if` clause when one is supplied, would it not be better to reference the `import` aspect instead?

**Review by Naumel — DISMISSED** [review]
Build is failing, otherwise change looks sane, only one nitpick as part of the review.

```
@aws-cdk/aws-events: lib/event-bus.ts:318:3 - error TS2376: A 'super' call must be the first statement in the constructor when a class contains initialized properties, parameter properties, or private identifiers.
@aws-cdk/aws-events: 318   constructor(scope: Construct, id: string, props?: EventBusProps) {
@aws-cdk/aws-events:       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
@aws-cdk/aws-events: 319     const { eventBusName, eventSourceName } = EventBus.eventBusProps(
@aws-cdk/aws-events:     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
@aws-cdk/aws-events: ... 
@aws-cdk/aws-events: 339     this.eventSourceName = eventBus.eventSourceName;
@aws-cdk/aws-events:     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
@aws-cdk/aws-events: 340   }
@aws-cdk/aws-events:     ~~~
@aws-cdk/aws-events: lib/event-bus.ts:374:3 - error TS2376: A 'super' call must be the first statement in the constructor when a class contains initialized properties, parameter properties, or private identifiers.
@aws-cdk/aws-events: 374   constructor(scope: Construct, id: string, attrs: EventBusAttributes) {
@aws-cdk/aws-events:       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
@aws-cdk/aws-events: 375     const arnParts = Stack.of(scope).splitArn(attrs.eventBusArn, ArnFormat.SLASH_RESOURCE_NAME);
@aws-cdk/aws-events:     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
@aws-cdk/aws-events: ... 
@aws-cdk/aws-events: 384     this.eventSourceName = attrs.eventSourceName;
@aws-cdk/aws-events:     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
@aws-cdk/aws-events: 385   }
@aws-cdk/aws-events:     ~~~
```

File: packages/@aws-cdk/aws-events/lib/event-bus.ts (hunk 1 above)
Comment (otaviomacedo) [reply packages/@aws-cdk/aws-events/lib/event-bus.ts]: Oh, you're absolutely right! There's no need for this property at all.

**Review by Naumel — APPROVED** [review]
Nice!

**Pull request merged**